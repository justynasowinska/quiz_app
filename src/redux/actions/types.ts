// questions
export const UPDATE_QUESTIONS = 'update_questions';

// progress
export const SET_NEXT_QUESTION = 'set_next_question';
export const RESET_PROGRESS = 'reset_progress';

// answers
export const ADD_USER_ANSWER = 'add_user_answer';
export const RESET_ANSWERS = 'reset_answers';