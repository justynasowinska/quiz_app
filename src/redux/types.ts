
export type DifficultlyType = 'hard' | 'medium' | 'easy';
export type BooleanCorrectAnswerType = 'True' | 'False';

export interface QuizQuestionType {
    category: string;
    type: string;
    difficulty: DifficultlyType;
    question: string;
    correct_answer: BooleanCorrectAnswerType; // maybe just string?
    incorrect_answers: Array<BooleanCorrectAnswerType>;
}

export type QuizQuestionsStateType = Array<QuizQuestionType>;

export interface QuizProgressType {
    currentQuestion: number;
}

export interface QuizAnswerType {
    questionId: number;
    answer: BooleanCorrectAnswerType;
}

export type QuizAnswersType = Array<QuizAnswerType>;