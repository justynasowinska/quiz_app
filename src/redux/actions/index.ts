export * from './answers';
export * from './progress';
export * from './questions';