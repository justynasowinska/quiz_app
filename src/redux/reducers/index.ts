import { combineReducers } from 'redux';

import quizDataReducer from './quizQuestionsReducer';
import quizProgressReducer from './quizProgressReducer';
import quizAnswersReducer from './quizAnswersReducer';

export const rootReducer = combineReducers({
    questions: quizDataReducer,
    progress: quizProgressReducer,
    answers: quizAnswersReducer
});

export default rootReducer;