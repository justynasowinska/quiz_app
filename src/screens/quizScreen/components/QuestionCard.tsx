import * as React from 'react';
import { View, StyleSheet, Text } from 'react-native';

import { THEME_COLORS } from '../../../config/colors';
import { QuizQuestionType } from '../../../redux/types';
import { decodeWithEntities, getDifficultyLevelColor } from '../../../utils/helpers';

interface PropTypes {
    question: QuizQuestionType;
}

export const QuestionCard = (props: PropTypes) => {
    const { question: { difficulty, question } } = props;
    const difficultyColor = getDifficultyLevelColor(difficulty);

    return (
        <View style={[styles.card, { borderColor: difficultyColor }]}>
            <Text style={styles.questionText}>
                {decodeWithEntities(question)}
            </Text>
        </View>
    );
};

const styles = StyleSheet.create({
    card: {
        width: '100%',
        aspectRatio: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: THEME_COLORS.white,
        borderWidth: 1,
        padding: 20
    },
    questionText: {
        fontSize: 20,
        color: THEME_COLORS.appTextColor,
        textAlign: 'center'
    }
});