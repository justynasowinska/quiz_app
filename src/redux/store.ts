import { createStore, applyMiddleware } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/es/stateReconciler/autoMergeLevel2';
import storage from 'redux-persist/lib/storage';

import reducers from './reducers';

const persistConfig = {
    key: 'root',
    storage,
    stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, reducers);

const middlewares = [];

if (__DEV__) {
    const { logger } = require('redux-logger');

    middlewares.push(logger);
}

export const store = createStore(persistedReducer, applyMiddleware(...middlewares));
export const persistor = persistStore(store);