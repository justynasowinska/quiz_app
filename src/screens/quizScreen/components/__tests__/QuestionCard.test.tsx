import * as React from 'react';
import { shallow } from 'enzyme';

import { QuestionCard } from '../QuestionCard';
import { QuizQuestionType } from '../../../../redux/types';

describe('Quiz Screen QuestionCard Component', () => {
    it('matches to the snapshot', () => {
        const question: QuizQuestionType = {
            category: 'Entertainment: Video Games',
            type: 'boolean',
            difficulty: 'hard',
            question: 'Unturned originally started as a Roblox game.',
            correct_answer: 'True',
            incorrect_answers: ['False']
        };

        const component = shallow(<QuestionCard question={question} />);

        expect(component).toMatchSnapshot();
    });
});
