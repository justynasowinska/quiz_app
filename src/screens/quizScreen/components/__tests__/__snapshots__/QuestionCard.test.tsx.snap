// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Screen QuestionCard Component matches to the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "#fff",
        "borderWidth": 1,
        "justifyContent": "center",
        "padding": 20,
        "width": "100%",
      },
      Object {
        "borderColor": "#CD0720",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#474D54",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    Unturned originally started as a Roblox game.
  </Text>
</View>
`;
